### PR TITLE
Commit Cargo.lock to repository

### PR DESCRIPTION
**Pull request by kpcyrd**
Related to #345

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -122,6 +122,17 @@ jobs:
       - run: make src/rustls.h
       - run: git diff --exit-code
 
+  audit:
+    name: Audit Cargo.lock for vulnerable dependencies
```
[the remark this answers is not in the document]
Comment (kpcyrd) [reply .github/workflows/test.yaml]: Done!

**Comment by djc** [comment]
I think you'll want to make sure CI jobs use `--locked` to that the `Cargo.lock` is actually getting tested.

**Review by cpu — APPROVED** [review]
This looks good to me and seems in keeping with [the updated guidance from the Rust project](https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html) on committing lockfiles :+1:

**Comment by kpcyrd** [comment]
I've added `--locked` in multiple places, if `Cargo.lock` is compatible with `Cargo.toml` it should be picked up automatically, if it's out-of-sync multiple jobs would now start failing (including `make`).

File: Makefile [hunk 2]
```diff
@@ -5,6 +5,7 @@ else
 endif
 
 CARGO ?= cargo
+CARGOFLAGS += --locked
```
Comment (cpu) [inline Makefile:8]: I think the `Makefile.Windows` and `CMakeLists.txt` files might need similar updates.

Comment (kpcyrd) [reply Makefile:8]: Done

**Review by cpu — APPROVED** [review]

**Comment by cpu** [comment]
@jsha Do you have any reservations about checking in the `Cargo.lock` for this project?

**Review by jsha — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**